web_extension/content/scripts/new_tab.js

```diff
@@ -1,14 +1,14 @@
 'use strict';
 
-async function fetchImage() {
-  const feed_url = options.feed.url
-  const response = await fetch(feed_url)
-  const feed = await response.json()
-  const number = parseInt(Math.random() * feed.items.length)
-  const item = feed.items[number]
-
-  set_image(item.url)
-  show_info(item)
+async function chooseImage() {
+  let item = await PhotoChooser.pick()
+  if (item) {
```

Comment:
```suggestion
  if (!item) return
```

Reply:
Seems legit but if I take your suggestion the code won’t work anymore...the diff doesn’t produce valid codes

Reply:
also I miss ruby